freedomofpress/securedrop-client
# update reply sender name during sync if it has changed

# Description

Towards https://github.com/freedomofpress/securedrop-client/issues/76

This depends on https://github.com/freedomofpress/securedrop-sdk/pull/125 which can be installed via `pip install --upgrade git+https://github.com/freedomofpress/securedrop-sdk@add-journalist-name-to-reply#egg=securedrop-sdk` but should not be merged until the sdk is updated.

# Test Plan

1. `pip install git+https://github.com/freedomofpress/securedrop-sdk@add-journalist-name-to-reply#egg=securedrop-sdk`
2. run `make test` and see all tests pass locally (they won't pass here until the sdk is updated)
3. log into the client
4. log into the journalist UI with a new journalist account
5. send a reply to a source as the new journalist.
6. wait until next sync then check local client db
   - [ ] verify local db shows an account for that new journalist in the users table
7. change first and last name of this new journalist account from the journalist UI
8. wait until next sync then check local client db
   - [ ] verify local db shows the new first and last names for the journalist in the users table
9. change first and last name to empty string from the journalist UI
10. wait until next sync then check local client db
    - [ ] verify local db shows updated first and last name
11. change username of the new journalist account from the admin UI
12. wait until next sync then check local client db
    - [ ] verify local db shows updated username
